components/org.wso2.carbon.identity.context.rewrite.valve/src/main/java/org/wso2/carbon/identity/context/rewrite/valve/TenantContextRewriteValve.java

```diff
@@ -156,6 +160,11 @@ public void invoke(Request request, Response response) throws IOException, Servl
         return rewriteContexts;
     }
 
+    private boolean isTenantQualifiedUrlsEnabled() {
```

Comment:
This needs to be called at the valve init method , assign to object variable and then use at line 84. Reading the config for every request is expensive in performance side.

Reply:
As @IsuraD suggested, this check was removed